vars/mavenExecute.groovy

```diff
@@ -10,7 +10,11 @@ def call(Map parameters = [:]) {
     final script = checkScript(this, parameters) ?: this
     parameters = DownloadCacheUtils.injectDownloadCacheInMavenParameters(script, parameters)
 
-    List credentials = [ ]
+    validateParameter(parameters.defines, 'defines')
+    validateParameter(parameters.flags, 'flags')
+    validateParameter(parameters.goals, 'goals')
```

Comment:
Actually all parameters need to be validated which may be quoted, i.e. those which contain paths. These are not list parameters: pomPath, projectSettingsFile, globalSettingsFile and m2Path, so another method to check just a string for quotes is probably necessary.